topics/reference_guide/custom_language_support/websymbols_integration.md

```diff
@@ -0,0 +1,309 @@
+- `CssDeclaration` - Web Symbols should represent available CSS properties for matching. *The CSS declaration actual name should not be taken into account when building the scope*.
+
+#### Pseudo-elements {#css-pseudo-elements}
+
+|---|---|
+|||
+|Namespace|`css`|
+|Kind|`pseudo-elements`|
+
+Web Symbols, which represent available [CSS pseudo-elements](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements).
+Symbols names *should not* be prefixed with `::`.
+CSS pseudo-elements can be contributed statically or dynamically (through `WebSymbolsQueryConfigurator`).
+The statically contributed symbols are available globally, depending on the context setting of the contributing Web Types file.
+
+The additional scope for matching pseudo-elements is built from Web Symbols matching HTML element name preceding the pseudo-element keyword.
+
+For dynamic contributions the `WebSymbolsQueryConfigurator#getScope`'s `element` parameter can be:
+- `CssPseudoSelector` - Web Symbols should represent available CSS pseudo-elements for matching. *The CSS pseudo-element actual name should not be taken into account when building the scope*.
+
+Dedicated support for `WebSymbol` interface properties:
+- `properties[PROP_ARGUMENTS]` - `true` if pseudo-element keyword accepts arguments.
+
+#### Pseudo-classes {#css-pseudo-classes}
+
+|---|---|
+|||
+|Namespace|`css`|
+|Kind|`pseudo-classes`|
+
+Web Symbols, which represent available [CSS pseudo-classes](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes).
+Symbols names *should not* be prefixed with `:`.
+CSS pseudo-classes can be contributed statically or dynamically (through `WebSymbolsQueryConfigurator`).
+The statically contributed symbols are available globally, depending on the context setting of the contributing Web Types file.
+
+The additional scope for matching pseudo-classes is built from Web Symbols matching HTML element name preceding the pseudo-class keyword.
+
+For dynamic contributions the `WebSymbolsQueryConfigurator#getScope`'s `element` parameter can be:
+- `CssPseudoSelector` - Web Symbols should represent available CSS pseudo-classes for matching. *The CSS pseudo-class actual name should not be taken into account when building the scope*.
+
+Dedicated support for `WebSymbol` interface properties:
+- `properties[PROP_ARGUMENTS]` - `true` if pseudo-class keyword accepts arguments.
+
+#### Functions {#css-functions}
+
+|---|---|
+|||
+|Namespace|`css`|
+|Kind|`functions`|
+
+Web Symbols, which represent available [CSS functions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions).
+CSS functions can be contributed statically or dynamically (through `WebSymbolsQueryConfigurator`).
+The statically contributed symbols are available globally, depending on the context setting of the contributing Web Types file.
+
+The additional scope for matching functions is built from Web Symbols matching the CSS property name, which value is being calculated.
```

Comment:
The last part of this sentence sounds wrong, but I wonder if I understood the meaning correctly. Maybe:

```suggestion
The additional scope for matching functions is built from Web Symbols matching the CSS property name, the value of which is being calculated.
```